src/main/scala/org/allenai/wikitables/WikiTablesEntityLinker.scala

```diff
@@ -0,0 +1,207 @@
+
+    entityLinkings.toVector
+  }
+  
+  def getEntityLinking(example: WikiTablesExample): RawEntityLinking = {
+    val links = ListBuffer[(Option[Span], Formula)]()
+    val foundFormulas = MutableSet[Formula]()
+
+    EditDistanceFuzzyMatcher.opts.expandAbbreviations = true;
+    EditDistanceFuzzyMatcher.opts.fuzzyMatchSubstring = true;
+    EditDistanceFuzzyMatcher.opts.alsoReturnUnion = true;
+    EditDistanceFuzzyMatcher.opts.alsoMatchPart = true;
+    EditDistanceFuzzyMatcher.opts.fuzzyMatchMaxEditDistanceRatio = 0.3;
+
+    val matcher = new EditDistanceFuzzyMatcher(example.getContext().graph
+        .asInstanceOf[TableKnowledgeGraph]);
+
+    // We don't actually use the spans of formulas at the moment.
+    /*
+    HashMap<Formula, Pair<Integer, Integer>> formulasPresent = new HashMap<>();
+    for (int i=0; i <= exTokens.size()-1; i++) {
+      for (int j=i+1; j <= exTokens.size(); j++) {
+        if (j == i+1) {
+          // We're looking at a span of one token
+          String token = exTokens.get(i);
+          if (token.matches("[-+]?\\d*\\.?\\d+")) {
+            Formula formula = Formula.fromString(token);
+            Pair<Integer, Integer> span = new Pair<>(i, j);
+            entityLinking.add(new Pair<>(span, formula));
+            formulasPresent.put(formula, span);
+          }
+        }
+        Collection<Formula> linkedFormulas = matcher.getFuzzyMatchedFormulas(exTokens, i, j,
+                                         FuzzyMatchFnMode.ENTITY);
+        linkedFormulas.addAll(matcher.getFuzzyMatchedFormulas(exTokens, i, j,
+                FuzzyMatchFnMode.BINARY));
+        for (Formula formula: linkedFormulas) {
+          // TODO: Store all the spans in entity linking instead of just the shortest one.
+          Pair<Integer, Integer> span = new Pair<>(i, j);
+          if (formulasPresent.containsKey(formula)) {
+            Pair<Integer, Integer> previousSpan = formulasPresent.get(formula);
+            int previousSpanLength = previousSpan.getSecond() - previousSpan.getFirst();
+            int currentSpanLength = span.getSecond() - span.getFirst();
+            if (currentSpanLength < previousSpanLength) {
+              entityLinking.remove(new Pair<>(previousSpan, formula));
+              entityLinking.add(new Pair<>(span, formula));
+              formulasPresent.put(formula, span);
+            }
+          } else {
+            entityLinking.add(new Pair(span, formula));
+            formulasPresent.put(formula, span);
+          }
+        }
+      }
+    }
+    */
+
+    val nerFormulas = getNerEntityLinking(example.sentence
+        .getAnnotation(WikiTablesUtil.NER_ANNOTATION).asInstanceOf[List[List[String]]])
+    links ++= nerFormulas
+    foundFormulas ++= nerFormulas.map(_._2)
+    
+    val entityFormulas = matcher.getAllFormulas(FuzzyMatchFnMode.ENTITY).asScala
+    // TODO: these are unused, not sure why.
+    // val unaryFormulas = matcher.getAllFormulas(FuzzyMatchFnMode.UNARY).asScala
```

Comment:
For the record, after a conversation on Slack, we decided that the unary formulas do not need to be included because the parser's grammar contains table independent stuff like `(fb:type.object.type fb:type.row)` and binary formulas contain `fb:row.row.column_name`.